app/src/main/java/com/github/se/gomeet/ui/mainscreens/profile/AddFriend.kt

```diff
@@ -0,0 +1,218 @@
+                            userViewModel.follow(uid)
+                          }
+                          // Directly update the state for immediate reflection
+                          followedUsers =
+                              if (isFollowing) {
+                                followedUsers - uid
+                              } else {
+                                followedUsers + uid
+                              }
+                        }
+                      }
+                    }
+                  }
+            }
+          }
+        }
+      }
+}
+
+@Composable
+fun ProfileImageUser(
+    userId: String,
+    modifier: Modifier = Modifier,
+    defaultImageResId: Int = R.drawable.gomeet_logo
+) {
+  var profilePictureUrl by remember { mutableStateOf<String?>(null) }
+
+  LaunchedEffect(userId) {
+    val db = FirebaseFirestore.getInstance()
+    val userDocRef = db.collection("users").document(userId)
+    try {
+      val snapshot = userDocRef.get().await()
+      profilePictureUrl = snapshot.getString("profilePicture")
+    } catch (e: Exception) {
+      e.printStackTrace()
+    }
+  }
+
+  Image(
+      painter =
+          if (!profilePictureUrl.isNullOrEmpty()) {
+            rememberAsyncImagePainter(profilePictureUrl)
+          } else {
+            painterResource(id = defaultImageResId)
+          },
+      contentDescription = "Profile picture",
+      modifier = modifier.size(40.dp).clip(CircleShape).background(color = Color.Gray),
+      contentScale = ContentScale.Crop)
+}
+
+@Composable
+fun UserItem(
+    user: GoMeetUser,
+    followedUsers: Set<String>,
+    nav: NavigationActions,
+    onFollowButtonClick: (String, Boolean) -> Unit
+) {
+  val isFollowing = followedUsers.contains(user.uid)
+
+  Row(
+      modifier =
+          Modifier.fillMaxWidth().padding(vertical = 3.dp).clickable {
+            nav.navigateToScreen(Route.OTHERS_PROFILE.replace("{uid}", user.uid))
+          },
+      verticalAlignment = Alignment.CenterVertically,
+      horizontalArrangement = Arrangement.SpaceBetween) {
+        Row(verticalAlignment = Alignment.CenterVertically) {
+          ProfileImageUser(userId = user.uid)
+          Spacer(modifier = Modifier.width(16.dp))
+          Column {
+            Text(
+                text = "${user.firstName} ${user.lastName}",
+                style = MaterialTheme.typography.bodyMedium)
+            Text(text = user.username, style = MaterialTheme.typography.bodySmall)
+          }
+        }
+        Button(
+            colors =
+                ButtonDefaults.buttonColors(
+                    if (isFollowing) MaterialTheme.colorScheme.outlineVariant else Color.White,
+                    if (isFollowing) MaterialTheme.colorScheme.onSecondary
+                    else MaterialTheme.colorScheme.tertiary),
+            onClick = { onFollowButtonClick(user.uid, isFollowing) }) {
+              Text(text = if (isFollowing) "Unfollow" else "Follow")
+            }
```

Comment:
Button shape, size and colors doesn't match the buttons in Followers / Following screens.

For reference, here is the Button settings in thos screens :
Button(
                    shape = RoundedCornerShape(10.dp),
                    onClick = {
                      userViewModel.unfollow(user.uid)
                      isFollowing = false
                    },
                    modifier =
                        Modifier.padding(start = 15.dp).width(110.dp).testTag("UnfollowButton"),
                    colors =
                        ButtonColors(
                            containerColor = MaterialTheme.colorScheme.primaryContainer,
                            contentColor = Color.Black,
                            disabledContentColor = Color.Transparent,
                            disabledContainerColor = Color.Transparent)) {
                      Text(text =  insert your text, style = MaterialTheme.typography.labelLarge)
                    }
                    
Here is a picture demonstrating these differences :
<img width="102" alt="Capture d’écran 2024-05-15 à 18 39 02" src="https://github.com/SwEnt-Project-G18/SwEntApp/assets/92448776/e989d353-69b2-49b3-83b5-f4190882171c">
<img width="121" alt="Capture d’écran 2024-05-15 à 18 39 21" src="https://github.com/SwEnt-Project-G18/SwEntApp/assets/92448776/465a6ea7-7c21-4238-8260-5066e364890c">

Reply:
Fixed. Thanks for the feedback!